manimlib/extract_scene.py

```diff
@@ -142,35 +143,48 @@ def get_indent(code_lines: list[str], line_number: int) -> str:
     return n_spaces * " "
 
 
-def insert_embed_line_to_module(module: Module, line_number: int):
+def insert_embed_line_to_module(module: Module, run_config: Dict) -> None:
     """
     This is hacky, but convenient. When user includes the argument "-e", it will try
     to recreate a file that inserts the line `self.embed()` into the end of the scene's
     construct method. If there is an argument passed in, it will insert the line after
     the last line in the sourcefile which includes that string.
     """
     lines = inspect.getsource(module).splitlines()
+    line_number = run_config.embed_line
 
     # Add the relevant embed line to the code
     indent = get_indent(lines, line_number)
     lines.insert(line_number, indent + "self.embed()")
     new_code = "\n".join(lines)
 
+    # When the user executes the `-e <line_number>` command,
+    # it should automatically identifies the nearest class
+    # defined above `<line_number>` as 'scene_names'.
+    classes = list(filter(lambda line: line.startswith("class"), lines[:line_number]))
```

Comment:
What about classes that are defined as nested class, i.e. inside some other classes? They would have an indentation so `line.startswith("class")` would return `False`.

Reply:
I guess the Scene Class (like: `class SomeName(Scene)`) would not be the nested one. So, I think in that case as well, it works.